### ISSUE TITLE
Add predicted functional scores to `get_mutations_df`

### ISSUE THREAD

**Issue by jgallowa07** [matsengrp/multidms#188]
# Add predicted functional scores to `get_mutations_df`

## Summary

Extend `Model.get_mutations_df` to always include `predicted_func_score_{condition}` columns reporting the model's predicted functional score for each single mutation on its condition-specific wild-type background. This gives users direct access to per-mutation predicted phenotypes alongside the existing `beta` and `shift` parameters.

## Background

`Model.get_mutations_df` currently returns a DataFrame indexed by mutation with columns for:
- `beta_{condition}` (latent-space mutation effects)
- `shift_{condition}` (difference from reference betas, for non-reference conditions)
- `times_seen_{condition}` (counts from training data)

However, users frequently want the **predicted functional score** for each single mutation---i.e., the quantity that passes through the global epistasis function and calibration scaling. This is especially important for Sigmoid models where the relationship between latent `beta` values and observed functional scores is nonlinear. Today users must manually construct single-mutant DataFrames and call `add_phenotypes_to_df` to obtain these values, which is cumbersome.

## Proposed Approach

For each condition, build a `jaxmodels.Data` object whose variant matrix `X` encodes every known single mutation on its correct condition-specific background. Use the existing `Data.single_mut_encodings` cached property, which already handles the critical distinction between reference-sequence conditions (identity matrix) and non-reference conditions (identity + bundle mutations at non-identical sites). Then call `Model._jax_model.predict_score` per condition and append `predicted_func_score_{condition}` columns.

**Key design decisions:**

1. **Always include predictions** rather than gating behind a flag. The computation is cheap (one matrix multiply per condition on a small identity-like matrix) and the columns are universally useful.

2. **Leverage `Data.single_mut_encodings`** for correct per-condition encoding. For non-reference conditions with different WT sequences, a single mutation's encoding must include 1s at bundle mutation indices (non-identical sites). `Data.single_mut_encodings` already handles this via `convert_subs_wrt_ref_seq` + `sub_str_to_binary`, so we reuse it rather than reimplementing the logic.

3. **Add a private helper `_get_single_mutation_data`** that wraps the encoding into a `jaxmodels.Data` object for a given condition. This keeps `get_mutations_df` clean and makes the prediction logic independently testable.

### Programmatic API

```python
# After fitting a model:
mutations_df = model.get_mutations_df(times_seen_threshold=3)

# DataFrame now includes (in addition to existing columns):
#   predicted_func_score_a, predicted_func_score_b, ...
mutations_df[["beta_a", "predicted_func_score_a", "shift_b", "predicted_func_score_b"]]
```

The return type and existing columns are unchanged. New columns are appended.

### 1. New private helper: `Model._get_single_mutation_data`

**File:** `multidms/model.py` (existing)

Add a private method that builds a `jaxmodels.Data` object encoding every known single mutation for a given condition, using the existing `Data.single_mut_encodings`.

```python
def _get_single_mutation_data(self, condition: str) -> jaxmodels.Data:
    """Build jaxmodels.Data encoding each single mutation as a separate variant.

    For reference-sequence conditions, X is the identity matrix (each row
    encodes one mutation). For non-reference conditions with different WT
    sequences, each row also includes 1s at bundle mutation indices
    (non-identical sites), matching the encoding used by
    ``_encode_variants`` and ``add_phenotypes_to_df``.

    Uses ``Data.single_mut_encodings`` which handles this distinction.
    """
    X = self._data.single_mut_encodings[condition]
    x_wt = self._jax_data_sets[condition].x_wt
    n_mutations = len(self._data.mutations)
    functional_scores = np.zeros(n_mutations)

    return jaxmodels.Data(
        x_wt=x_wt,
        X=X,
        functional_scores=functional_scores,
    )
```

**Rationale:** `Data.single_mut_encodings` is a cached property that already correctly handles the encoding for both reference and non-reference conditions:
- **Reference-sequence conditions**: returns the identity matrix (row `i` has a single 1 at index `i`)
- **Non-reference conditions**: calls `convert_subs_wrt_ref_seq` on each mutation, then `sub_str_to_binary`, producing rows with 1s at both the mutation index *and* all bundle mutation indices (non-identical sites between the condition and reference WT)

This ensures predictions are consistent with `_encode_variants` and `add_phenotypes_to_df`.

### 2. Update `Model.get_mutations_df`

**File:** `multidms/model.py` (existing)

After the existing beta/shift column logic, add a block that computes and appends predicted functional scores.

```python
# After existing beta/shift columns are added...

# Add predicted functional score columns
for condition in self._data.conditions:
    single_mut_data = self._get_single_mutation_data(condition)
    predictions = self._jax_model.predict_score({condition: single_mut_data})
    mutations_df[f"predicted_func_score_{condition}"] = np.array(
        predictions[condition]
    )
```

### 3. Update docstring

Update the `get_mutations_df` docstring to document the new columns:

```
Returns
-------
pd.DataFrame
    DataFrame with mutations as rows (index) and columns:
    - beta_{condition} for each condition
    - shift_{condition} for each non-reference condition
    - predicted_func_score_{condition} for each condition
```

### Single mutation encoding

The prediction formula for each single mutation is:

```
predicted_func_score = alpha * (g(phi(X_i)) - g(phi(x_wt)))
```

where `phi(X_i) = beta0 + X_i @ beta` and `X_i` is the binary encoding of the single-mutant variant.

### Why the identity matrix alone is insufficient

For **reference-sequence conditions**, X is the identity matrix: row `i` has a single 1 at column `i`, encoding only mutation `i`. This is correct because the reference WT is encoded as all-zeros.

For **non-reference conditions** with different WT sequences, a single mutation like `M1A` in condition `b` must be encoded with 1s at both the `M1A` column *and* all bundle mutation columns (non-identical sites between condition `b`'s WT and the reference WT). Without the bundle, the encoding would represent a variant that has mutation `M1A` but *also* reverts the condition's WT back to the reference at all non-identical sites---which is not the intended single-mutant background.

`Data.single_mut_encodings` handles this correctly by calling `convert_subs_wrt_ref_seq` (which appends bundle mutations) followed by `sub_str_to_binary` for non-reference conditions.

### Consistency with `add_phenotypes_to_df`

The `_encode_variants` method used by `add_phenotypes_to_df` performs the same conversion: for non-reference conditions, it calls `convert_subs_wrt_ref_seq` before encoding. Using `Data.single_mut_encodings` ensures that `predicted_func_score_{condition}` from `get_mutations_df` matches what `add_phenotypes_to_df` would return for the same single-mutant variants.

### Performance

The computation involves one sparse matrix multiply of shape `(n_mutations, n_mutations)` per condition. For typical DMS datasets with ~4000 mutations and 2-5 conditions, this is negligible (< 100ms total).

## Testing Strategy

- **Doctest in `get_mutations_df`**: Verify `predicted_func_score_{condition}` columns appear in output.
- **Doctest in `_get_single_mutation_data`**: Verify returned Data object has correct shape.
- **Unit test: Identity model consistency**: For a model with `ge_type='Identity'`, verify that `predicted_func_score_{condition}[i]` equals `alpha * beta[i]` for the reference condition (since `g` is identity and WT has all-zero encoding relative to itself).
- **Unit test: Round-trip with `add_phenotypes_to_df`**: Construct a single-mutant DataFrame, predict with `add_phenotypes_to_df`, and verify results match `predicted_func_score_{condition}` from `get_mutations_df`.
- **Edge cases:**
  - Model with only one condition (reference only, no shifts)
  - Non-reference condition with different WT sequence (bundle mutations)
  - Mutations with `times_seen=0` in some conditions (should still have predictions)
  - Filtering with `times_seen_threshold` (predictions should be filtered along with other columns)

## Documentation Updates

- [ ] Update `get_mutations_df` docstring (columns list, example)
- [ ] Update any notebook examples that display `get_mutations_df` output

## Dependencies

No new dependencies. Uses existing `scipy.sparse`, `jax.experimental.sparse`, and `jaxmodels` infrastructure.

### Alternative 1: Optional `include_predictions` flag

Add `include_predictions: bool = False` parameter to `get_mutations_df`.

**Why not chosen:** The computation is cheap and the columns are universally useful. An optional flag adds API complexity without meaningful benefit. Users who don't want the columns can simply drop them.

### Alternative 2: Separate `get_mutations_predicted_df` method

Create a new method that extends `get_mutations_df` with predictions.

**Why not chosen:** Creates a confusing dual-method API. Users would need to decide which method to call, and the predicted version would always be preferable. Better to have one method that returns complete information.

### Alternative 3: Reuse `_encode_variants` with a synthetic DataFrame

Build a DataFrame of all single mutations and pass it through the existing `_encode_variants` method.

**Why not chosen:** While `_encode_variants` handles the full encoding pipeline including `convert_subs_wrt_ref_seq`, it requires constructing a DataFrame with condition and substitution string columns, then grouping by condition. `Data.single_mut_encodings` is more direct---it is already cached and produces the correct per-condition encoding matrices.

### Alternative 4: Naive identity matrix for all conditions

Use `scipy.sparse.eye(n_mutations)` as X for every condition.

**Why not chosen:** This is incorrect for non-reference conditions with different WT sequences. Those conditions require bundle mutations (non-identical sites) to be included in each single mutation's encoding. Without the bundle, predictions would represent variants that both carry the mutation *and* revert the condition's WT to the reference---not the intended single-mutant background.

## Success Criteria

- [ ] `get_mutations_df` returns `predicted_func_score_{condition}` columns for all conditions
- [ ] For Identity GE models, `predicted_func_score` equals `alpha * beta` for the reference condition
- [ ] Predictions match `add_phenotypes_to_df` on equivalent single-mutant DataFrames
- [ ] All existing doctests continue to pass
- [ ] `times_seen_threshold` filtering applies consistently to new columns

## Future Work

- Add `predicted_latent_{condition}` and `predicted_fitness_{condition}` columns if users request them (same infrastructure, just extract intermediate values)
- Consider caching single-mutation predictions if they become a bottleneck for interactive use
